bin/oref0-pump-loop.sh

```diff
@@ -92,9 +92,42 @@ main() {
             # pump-loop errored out for some reason
             fail "$@"
         fi
+        run_plugins
     fi
 }
 
+function run_script() {
+  file=$1
+
+  # only run plugin script files if they have the executable bit set
+  if [[ -x "$file" ]]
+  then
+    echo "Running plugin script ($file)... "
+    timeout 60 $file
+    echo "Completed plugin script ($file). "
+
+    # -d means to only run the script once, so remove it once run
+    if [[ "$2" == "-d" ]]
+    then
+      #echo "Removing script file ($file)"
+      rm $file
+    fi
+  fi
+}
+
+
+function run_plugins {
+        once=plugins/once
+        every=plugins/every
+        mkdir -p $once
+        mkdir -p $every
+        echo "scripts placed in this directory will run once after curent loop and be removed" > $once/readme.txt
+        echo "scripts placed in this directory will run after every loop" > $every/readme.txt
+        find $once/* | while read file; do run_script "$file" -d ; done
+        find $every/* | while read file; do run_script "$file" ; done
```

Comment:
I would recommend adding `-executable` to the find commands to eliminate the need for the `if` block in run_plugins testing for execution permissions.  It makes run_plugins a little cleaner.
```
find $once/* -executable | while read file; do run_script "$file" -d ; done
find $every/* -executable | while read file; do run_script "$file" ; done
```